### PR TITLE
Upgrade xgs SAI to 10.1.0.0

### PR DESCRIPTION
**Pull request by Gfrom2016**
#### Why I did it
Upgrade xgs SAI to 10.1 version.

##### Work item tracking
- Microsoft ADO **(number only)**: 25931321

#### How I did it
Upgrade xgs SAI version in sai.mk file.

#### How to verify it
Run full qualification on 7050cx3/7260cx3:
- 7050cx3:
  -  https://dev.azure.com/mssonic/internal/_build/results?buildId=425450&view=results
  -  https://dev.azure.com/mssonic/internal/_build/results?buildId=425449&view=results
- 7260cx3: https://elastictest.org/scheduler/testplan/656f2b2b617fb27e41557494?leftSideViewMode=detail&prop=status&order=ascending

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305
- [x] 202311

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: platform/broadcom/sai.mk [hunk 1]
```diff
@@ -1,6 +1,6 @@
-LIBSAIBCM_XGS_VERSION = 8.4.21.0
+LIBSAIBCM_XGS_VERSION = 10.1.0.0
 LIBSAIBCM_DNX_VERSION = 7.1.111.1
-LIBSAIBCM_XGS_BRANCH_NAME = SAI_8.4.0_GA
+LIBSAIBCM_XGS_BRANCH_NAME = zitingguo/enable_10.1.0
```
Comment (StormLiangMS) [inline platform/broadcom/sai.mk:3]: should we use SAI_10.1.0_GA?

**Review by StormLiangMS — DISMISSED** [review]

**Comment by StormLiangMS** [comment]
@Gfrom2016 there is a conflict

File: platform/broadcom/sai.mk (hunk 1 above)
Comment (Gfrom2016) [reply platform/broadcom/sai.mk]: Yes, it should be SAI_10.1.0_GA. Fixed.

**Comment by Gfrom2016** [comment]
> @Gfrom2016 there is a conflict



> @Gfrom2016 there is a conflict

conflict resolved.

**Review by StormLiangMS — APPROVED** [review]
LGTM

**Pull request merged**